extractor/src/main/java/org/schabi/newpipe/extractor/stream/AudioStream.java

```diff
@@ -4,111 +4,380 @@
  * Created by Christian Schabesberger on 04.03.16.
  *
  * Copyright (C) Christian Schabesberger 2016 <chris.schabesberger@mailbox.org>
- * AudioStream.java is part of NewPipe.
+ * AudioStream.java is part of NewPipe Extractor.
  *
- * NewPipe is free software: you can redistribute it and/or modify
+ * NewPipe Extractor is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation, either version 3 of the License, or
  * (at your option) any later version.
  *
- * NewPipe is distributed in the hope that it will be useful,
+ * NewPipe Extractor is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
  * GNU General Public License for more details.
  *
  * You should have received a copy of the GNU General Public License
- * along with NewPipe.  If not, see <http://www.gnu.org/licenses/>.
+ * along with NewPipe Extractor. If not, see <https://www.gnu.org/licenses/>.
  */
 
 import org.schabi.newpipe.extractor.MediaFormat;
 import org.schabi.newpipe.extractor.services.youtube.ItagItem;
 
-public class AudioStream extends Stream {
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public final class AudioStream extends Stream {
```

Comment:
This class is final because of Checkstyle (this comment also applies for `VideoStream` and `SubtitlesStream`).

Reply:
> because of Checkstyle

Okay, can you maybe describe this a bit more accurate?

Reply:
Checkstyle requests the class to be final because it only has a private constructor, and hence it wouldn't make sense to extend it. Should this be changed?